lib/fpm/package/p5p.rb

```diff
@@ -0,0 +1,116 @@
+require "erb"
+require "fpm/namespace"
+require "fpm/package"
+require "fpm/errors"
+require "fpm/util"
+
+class FPM::Package::P5P < FPM::Package
+
+  option "--user", "USER",
+    "Set the user to USER in the prototype files.",
+    :default => 'root'
+
+  option "--group", "GROUP",
+    "Set the group to GROUP in the prototype file.",
+    :default => 'root'
+
+  option "--zonetype", "ZONETYPE",
+    "Set the allowed zone types (global, nonglobal, both)",
+    :default => 'value=global value=nonglobal' do |value|
+      case @value
+      when "both"
+        value = "value=global value=nonglobal"
+      else
+        value = "value=#{value}"
+      end
+    end # value
+
+  option "--publisher", "PUBLISHER", 
+    "Set the publisher name for the repository",
+    :default => 'FPM'
+
+  option "--lint" , :flag, "Check manifest with pkglint",
+    :default => true
+
+  option "--validate", :flag, "Validate with pkg install",
+    :default => true
+
+  def architecture
+    case @architecture
+    when nil, "native"
+      @architecture = %x{uname -p}.chomp
+    when "all"
+      @architecture = 'i386 value=sparc'
+    end
+
+    return @architecture
+  end # def architecture
+
+  def output(output_path)
+    
+    # Fixup the category to an acceptable solaris category
+    case @category
+    when nil, "default"
+      @category = 'Applications/System Utilities'
+    end
+
+    # Generate a package manifest.
+    pkg_generate = safesystemout("pkgsend", "generate", "#{staging_path}")
+    File.open("#{build_path}/#{name}.p5m.1", "w") do |manifest|
+      manifest.puts pkg_generate
+    end
+    safesystem("cp", "#{build_path}/#{name}.p5m.1", "#{build_path}/#{name}.p5m")
```

Comment:
The .p5m is the extension used to designate a package manifest (I assume thats akin to the specfile for rpm's).  The .1, .2, ... are progressions in building the final manifest and I left them behind as debug artifacts because it makes it easier to figure out where the process went wrong.